plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,403 @@
+                    }
+                }
+                batchPredictionListener.onResponse(null);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                batchPredictionListener.onFailure(e);
+            }
+        });
+
+    }
+
+    private void getMappedModelInputFromDocuments(
+        IngestDocument ingestDocument,
+        Map<String, String> modelParameters,
+        String originalFieldName,
+        String modelInputFieldName
+    ) {
+        String originalFieldPath = getFieldPath(ingestDocument, originalFieldName);
+        if (originalFieldPath != null) {
+            Object originalFieldValue = ingestDocument.getFieldValue(originalFieldPath, Object.class);
+            String originalFieldValueAsString = toString(originalFieldValue);
+            updateModelParameters(modelInputFieldName, originalFieldValueAsString, modelParameters);
+        }
+        // check for nested array
+        else {
+            if (originalFieldName.contains(DOT_SYMBOL)) {
+
+                Map<String, Object> sourceObject = ingestDocument.getSourceAndMetadata();
+
+                ArrayList<Object> fieldValueList = JsonPath
+                    .using(suppressExceptionConfiguration)
+                    .parse(sourceObject)
+                    .read(originalFieldName);
+
+                String originalFieldValueAsString = toString(fieldValueList);
+                updateModelParameters(modelInputFieldName, originalFieldValueAsString, modelParameters);
+            }
+        }
+    }
+
+    /**
+     * support multiple document fields to map to the same model input fields,
+     * check if the key existed, then append to existed value to be a list.
+     */
+    private void updateModelParameters(String modelInputFieldName, String originalFieldValueAsString, Map<String, String> modelParameters) {
+
+        if (modelParameters.containsKey(modelInputFieldName)) {
+            Object existingValue = modelParameters.get(modelInputFieldName);
+            List<Object> updatedList = (List<Object>) existingValue;
+            updatedList.add(originalFieldValueAsString);
+            modelParameters.put(modelInputFieldName, toString(updatedList));
+        } else {
+            modelParameters.put(modelInputFieldName, originalFieldValueAsString);
+        }
+
+    }
+
+    private String getFieldPath(IngestDocument ingestDocument, String originalFieldName) {
+        final boolean fieldPathIsNullOrEmpty = Strings.isNullOrEmpty(originalFieldName);
+
+        if (fieldPathIsNullOrEmpty) {
+            return null;
+        }
+        final boolean hasFieldPath = ingestDocument.hasField(originalFieldName, true);
+        if (!hasFieldPath) {
+            return null;
+        }
+        return originalFieldName;
+    }
+
+    /**
+     * appendFieldValue to the ingestDocument without changing the source
```

Comment:
Same comment, you don't add parameters list to java doc. Please check all of your java doc, make sure they are correct.

Reply:
added java doc to all methods